iceoryx_posh/include/iceoryx_posh/popo/modern_api/publisher_options.hpp

```diff
@@ -0,0 +1,34 @@
+// Copyright (c) 2020 by Apex.AI Inc. All rights reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#ifndef IOX_POSH_POPO_PUBLISHER_OPTIONS_HPP
+#define IOX_POSH_POPO_PUBLISHER_OPTIONS_HPP
+
+#include "iceoryx_posh/mepoo/memory_info.hpp"
+
+#include <cstdint>
+
+namespace iox
+{
+namespace popo
+{
+struct PublisherOptions
+{
+    uint64_t historyCapacity{0};
```

Comment:
I deliberatelly omitted the `m_` prefix and would suggest to not use it if the member is public. It's just weird to have the `m_` prefix in a public interface.

Reply:
I'd vote for sticking to our coding guidelines. If we decide otherwise, please document this exception in the `CONTRIBUTING.md`

Reply:
Aren't there already other m_ member in thinks like ChunkHeader that we already expose to the user?

Reply:
yes, at least in the ChunkHeader. I'm currently not aware of other structs where that's the case. I would also change it there and add it to the `CONTRIBUTING.md` if there are no objections

Reply:
@mossmaurice it's now or with the 2.0 release

Reply:
any other comments on this?

Reply:
@elfenpiff  I vote for getting rid of `m_` for public members. The gain is not present but the pain is!